OpenDreamRuntime/Procs/DMOpcodeHandlers.cs

```diff
@@ -525,11 +525,35 @@ static class DMOpcodeHandlers {
             return null;
         }
 
-        public static ProcStatus? BitOr(DMProcState state) {
-            int second = state.PopDreamValue().GetValueAsInteger();
-            int first = state.PopDreamValue().GetValueAsInteger();
+        public static ProcStatus? BitOr(DMProcState state) {                                // x | y
+            DreamValue secondValue = state.PopDreamValue();
 
-            state.Push(new DreamValue(first | second));
+            if (secondValue.TryGetValueAsInteger(out int secondInt)) {                      // x | 2
+                DreamValue firstValue = state.PopDreamValue();
+
+                if (firstValue.TryGetValueAsInteger(out int firstInt)) {                    // 1 | 2
+                    state.Push(new DreamValue(firstInt | secondInt));
+                } else if (firstValue.TryGetValueAsDreamList(out DreamList firstList)) {    // List | 2
```

Comment:
How this is done with other operators (`+`, `-`, etc) is there's an operator method in `IDreamMetaObject`. If the first value is a `DreamObject`, that method gets called. The override of `OperatorOr` in `DreamMetaObjectList` would handle the union.